ocaml/xapi/xapi_vm.ml

```diff
@@ -528,8 +528,12 @@ let clone ~__context ~vm ~new_name =
 (* We do call wait_in_line for snapshot and snapshot_with_quiesce because the locks are taken at *)
 (* the VBD level (with pause/unpause mechanism                                                   *)
 let snapshot ~__context ~vm ~new_name =
-	TaskHelper.set_cancellable ~__context;
-	Xapi_vm_snapshot.snapshot ~__context ~vm ~new_name
+	Local_work_queue.wait_in_line Local_work_queue.snapshot_vm_queue
```

Comment:
As noted in my comment below, the `OTHER_OPRTATION_IN_PROGRESS` error happens when snapshotting two separete VMs with VDIs on the same SR